web_services/foxx/brightid/index.js

```diff
@@ -194,156 +196,91 @@ const handlers = {
+      throw new errors.InvalidExpressionError(app.name, params.verification, err);
     }
     if (! verified) {
-      throw new errors.NotVerifiedError(contextId, appKey);
+      throw new errors.NotVerifiedError(params.app);
     }
 
-    let contextIds = db.getContextIdsByUser(coll, user);
-    if (contextId != contextIds[0]) {
-      unique = false;
+    if (! (module.context && module.context.configuration && module.context.configuration.wISchnorrPassword)){
+      throw new errors.WISchnorrPasswordNotSetError();
     }
 
-    if (timestamp == 'seconds') {
-      timestamp = parseInt(Date.now() / 1000);
-    } else if (timestamp == 'milliseconds') {
-      timestamp = Date.now();
-    } else {
-      timestamp = undefined;
-    }
-
-    // sign and return the verification
-    let sig, publicKey;
-    if (signed == 'nacl') {
-      if (! (module.context && module.context.configuration && module.context.configuration.publicKey && module.context.configuration.privateKey)){
-        throw new errors.KeypairNotSetError();
-      }
-
-      let message = appKey + ',' + contextIds.join(',');
-      if (timestamp) {
-        message = message + ',' + timestamp;
-      }
-      const privateKey = module.context.configuration.privateKey;
-      publicKey = module.context.configuration.publicKey;
-      sig = uInt8ArrayToB64(
-        Object.values(nacl.sign.detached(strToUint8Array(message), b64ToUint8Array(privateKey)))
-      );
-    } else if (signed == 'eth') {
-      if (! (module.context && module.context.configuration && module.context.configuration.ethPrivateKey)){
-        throw new errors.EthPrivatekeyNotSetError();
-      }
+    const password = module.context.configuration.wISchnorrPassword;
+    const server = new WISchnorrServer();
+    server.GenerateSchnorrKeypair(password);
 
-      let message, h;
-      if (context.idsAsHex) {
-        message = pad32(appKey) + contextIds.map(addressToBytes32).join('');
-      } else {
-        message = pad32(appKey) + contextIds.map(pad32).join('');
-      }
-      message = Buffer.from(message, 'binary').toString('hex');
-      if (timestamp) {
-        const t = timestamp.toString(16);
-        message += ('0'.repeat(64 - t.length) + t);
-      }
-      h = new Uint8Array(createKeccakHash('keccak256').update(message, 'hex').digest());
-      let ethPrivateKey = module.context.configuration.ethPrivateKey;
-      ethPrivateKey = new Uint8Array(Buffer.from(ethPrivateKey, 'hex'));
-      publicKey = Buffer.from(Object.values(secp256k1.publicKeyCreate(ethPrivateKey))).toString('hex');
-      const _sig = secp256k1.ecdsaSign(h, ethPrivateKey);
-      sig = {
-        r: Buffer.from(Object.values(_sig.signature.slice(0, 32))).toString('hex'),
-        s: Buffer.from(Object.values(_sig.signature.slice(32, 64))).toString('hex'),
-        v: _sig.recid + 27,
-      }
+    const q = { _key: id, roundedTimestamp: params.roundedTimestamp, app: params.app };
```

Comment:
I would include the original `info` object here--perhaps as a deterministic string and indexed in the DB to speed up the below query. `verification` should also be included here. If it might be useful to search by one of the common sub-elements of `info`, then `info` can be destructured like you have here, and consider adding indexes to any fields that might be searched.

Reply:
This was a bug and fixed now.